src/app/products/loan-products/loan-product-stepper/loan-product-interest-refund-step/loan-product-interest-refund-step.component.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {UntypedFormBuilder, UntypedFormGroup} from '@angular/forms';
+import {StringEnumOptionData} from './string-enum-option-data-model';
+
+@Component({
+  selector: 'mifosx-loan-product-interest-refund-step',
+  templateUrl: './loan-product-interest-refund-step.component.html',
+  styleUrls: ['./loan-product-interest-refund-step.component.scss']
+})
+export class LoanProductInterestRefundStepComponent implements OnInit {
+
+  @Input() loanProductsTemplate: any;
+  @Output() supportedInterestRefundTypes = new EventEmitter<StringEnumOptionData[]>();
+
+  loanProductInterestRefundForm: UntypedFormGroup;
+
+  supportedInterestRefundTypesOptions: StringEnumOptionData[];
+
+  constructor(private formBuilder: UntypedFormBuilder) {
+    this.createLoanProductInterestRefundForm();
+    this.setConditionalControls();
+  }
+
+  ngOnInit() {
+    this.supportedInterestRefundTypesOptions = this.loanProductsTemplate.supportedInterestRefundTypesOptions;
+    const values: StringEnumOptionData[] = this.loanProductsTemplate.supportedInterestRefundTypes;
+    const supportedInterestRefundTypes: string[] = this.mapStringEnumOptionToIdList(values);
+    this.loanProductInterestRefundForm.patchValue({
+      'supportedInterestRefundTypes': supportedInterestRefundTypes
+    });
+    this.supportedInterestRefundTypes.emit(values);
+  }
+
+  createLoanProductInterestRefundForm() {
+    this.loanProductInterestRefundForm = this.formBuilder.group({
+      'supportedInterestRefundTypes': ''
+    });
+  }
+
+  setConditionalControls() {
+    this.loanProductInterestRefundForm.get('supportedInterestRefundTypes').valueChanges
+      .subscribe(value => {
+        this.supportedInterestRefundTypes.emit(this.mapIdToStringEnumOptionList(value, this.loanProductsTemplate.supportedInterestRefundTypesOptions));
+      });
+  }
+
+  mapStringEnumOptionToIdList(incomingValues: StringEnumOptionData[]): string[] {
```

Comment:
Use the `OptionData` already defined in 'app/shared/models/option-data.model'

Reply:
Its not the same. OptionData id is number. StringEnumOptionData id is string